### DISCUSSION TITLE
Making sure GPU is being used

### DISCUSSION THREAD

**Discussion by DoubleF3lix** [openai/whisper#835] [Q&A]
Hello

I installed the package from the git repo, and then uninstalled torch and installed the CUDA 11.7 version. I also have CUDA 11.7 installed, and `torch.cuda.is_available()` returns `True`, however, in Task Manager, the Python process is only using about 1.5% of my GPU (a GTX 970) while my CPU is at about 15% usage (an i7-6700k). 

Is there another check I can do to make sure it is using the GPU like I'd expect?

**Comment by glangford** [comment]
Running from the command line with `--device cuda` has verbose logging that will tell you more (or call `transcribe()` with `verbose=True`). 

In general, see
[How do I check if PyTorch is using the GPU](https://stackoverflow.com/questions/48152674/how-do-i-check-if-pytorch-is-using-the-gpu)

A couple of other possibilities, fyi in case these past discussions are helpful. 
* https://github.com/openai/whisper/discussions/672 
* https://github.com/openai/whisper/discussions/47

**Reply by glangford** [reply]
One more thing - if your GPU can be used by Whisper, on paper it looks like the Medium model (for example) requires ~5GB VRAM and the GTX 970 spec says it has only 4GB.

**Reply by DoubleF3lix** [reply]
I'm running the small model, and I ran the `torch` checks to get the device name and it does identify my GTX 970, so I guess it is using it, just not a whole lot.
